cargo audit does not work on Rust 1.44 anymore

While running `cargo install cargo-audit --force` with Rust 1.44 we are getting following [error in smol_str](https://github.com/rust-analyzer/smol_str/issues/25):

```
error[E0658]: `while` is not allowed in a `const fn`
  --> C:\Users\serg\.cargo\registry\src\github.com-1ecc6299db9ec823\smol_str-0.1.17\src\lib.rs:58:9
   |
58 | /         while i < text.len() {
59 | |             buf[i] = text.as_bytes()[i];
60 | |             i += 1
61 | |         }
   | |_________^
   |
   = note: see issue #52000 <https://github.com/rust-lang/rust/issues/52000> for more information
```



`cargo-audit` uses `crates-index` which uses `smol_str` and now "cargo audit requires Rust 1.40 or later." claim does not hold.

our CI runs cargo audit and must keep running on Rust 1.44 due to performance issues with 1.45-1.46.

would it be possible to pin smol_str in Cargo.lock to something that works with older compiler versions?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,6 +28,7 @@ thiserror = "1"
 once_cell = "1.3"
 tempfile = "3"
 toml = "0.5"
+smol_str = "=0.1.16" # smol_str 0.1.17 is incompatible with Rust 1.44 or lower
 
 [dev-dependencies.abscissa_core]
 version = "0.5"

